src/apps/cacheStatus/options.cpp

```diff
@@ -223,16 +223,17 @@ void COptions::Init(void) {
 
 #ifndef HOST_NAME_MAX
 #define HOST_NAME_MAX 64
+#endif
+#ifndef LOGIN_NAME_MAX
 #define LOGIN_NAME_MAX 64
 #endif
     char hostname[HOST_NAME_MAX];
     gethostname(hostname, HOST_NAME_MAX);
-    char username[LOGIN_NAME_MAX];
-    getlogin_r(username, LOGIN_NAME_MAX);
-    if (isDockerMode()) {
-        memset(username, 0, LOGIN_NAME_MAX);
+    char username[LOGIN_NAME_MAX] = { 0 };
+    if (getlogin_r(username, LOGIN_NAME_MAX) != 0)
+        strncpy(username, "unknown", 8);
```

Comment:
Is "unknown" something people would expect to see? In my experience (admittedly a bit outdated) this value would have been "nobody". If we use "nobody" we can collapse this with the `isDockerMode` clause.

Reply:
Works for me. Only reason I put `unknown` is because that's more accurate -- when I observed this issue, it was running under a real user, but in a systemd service, which I suppose interfered with `getlogin_r` due to the lack of a TTY.